x-pack/legacy/plugins/task_manager/task_store.ts

```diff
@@ -137,6 +138,39 @@ export class TaskStore {
     return savedObjectToConcreteTaskInstance(savedObject);
   }
 
+  /**
+   * Reschedules a previously scheduled task.
+   *
+   * @param task - The task being rescheduled.
+   */
+  public async reschedule(
+    taskInstanceScheduling: TaskInstanceScheduling
+  ): Promise<ConcreteTaskInstance> {
+    const taskInstance = await this.getTask(taskInstanceScheduling.id);
+    return await this.update(
```

Comment:
I'm thinking this section may become a bit more tricky if we want to further reduce the possibility of getting 409 errors. We should evaluate if it's worth worrying about but I'll write some thoughts below.

Some tasks can change state very often and cause such errors. For example, it's possible a task could be updated from elsewhere between the `getTask` call and `update` call. Some of those changes include:
- Task moving from `idle` to `claiming`
- Task moving from `claiming` to `running`
- Task moving from `running` to `idle`

If we think this is worth pursuing, some options may include:
- Updating only the `interval` attribute in a separate call without the `version` attribute (guaranteed no 409s)
- Re-attempting a save if ever a 409 has been received (though this could be a repeated error, small odds)
- Some other logic about trying to update the `runAt`

Hopefully if we change things, the implementations are simple.

Reply:
Edit: I removed the `guaranteed no 409s` section from my comment as this is not true. They would only happen if ever at a low level a conflict is detected (since Lucene operation is a read then write). `retry_on_conflict` would be the next way around those.

Reply:
Yeah, this is what I'm still working on now... 
At the moment my blocker is I can't setup a reliable functional test that mimics this... so, it's on going.

I'm not yet sure what the ideal approach would be, but that does raise that we can't skip updating the `runAt` as that means we might end up with the same problem we have now which is that `interval` won't be applied until the next `runAt` expires if `get` gave us a task with `running` than then became `idle` by the time `update` happened.

This is why my instinct is actually a retry where we can reevaluate which fields to update. Thoughts?

Reply:
I've introduced a retry on version conflict - but it'll only retry twice, then it'll give up and bubble up the error.

What do you think?

Reply:
The changes you've done will definitely help reduce the odds of the task not getting an updated interval. I guess if ever it fails to attempt twice, the reschedule function throws an error? Which I think can be used to notify the user of the very small chance something went wrong and to try again their request again.

Reply:
Yes, exactly, if it fails after the second attempt the `reschedule` call will throw an error.
Whoever is using it will have to handle it.

In the case of the Alerting api that should bubble up to the api reply, but I'll double check.